Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
+```java
+public class MyParcelable implements Parcelable {
+     private int mData;
+
+     public int describeContents() {
+         return 0;
+     }
+
+     public void writeToParcel(Parcel out, int flags) {
+         out.writeInt(mData);
+     }
+
+     public static final Parcelable.Creator<MyParcelable> CREATOR
+             = new Parcelable.Creator<MyParcelable>() {
+         public MyParcelable createFromParcel(Parcel in) {
+             return new MyParcelable(in);
+         }
+
+         public MyParcelable[] newArray(int size) {
+             return new MyParcelable[size];
+         }
+     };
+     
+     private MyParcelable(Parcel in) {
+         mData = in.readInt();
+     }
+ }
+```
+
+Please note that `Parcelable` is not meant for storing data!
 
 #### Static Analysis
 
+In general: if the object persistence is used for persisting any sensitive information on the device, then make sure that the information is encrypted and signed/HMACed. See the chapters on data storage and cryptographic management for more details. Next, you need to make sure that obtaining the keys to decrypt and verify are only obtainable if the user is authenticated. Security checks should be made at the correct positions as defined in <sup>[13]</sup>.
+
+##### Object Serialization
 Search the source code for the following keywords:
 
 - `import java.io.Serializable`
 - `implements Serializable`
 
-Check if serialized data is stored temporarily or permanently within the app's data directory or external storage and if it contains sensitive data.
+##### JSON
+
+Static analysis depends on the library being used. In case of the need to counter memory-dumping, make sure that highly sensitive information is not stored in JSON as you cannot guarantee any anti-memory dumping techniques with the standard libraries. You can check for the following keywords per library:
+
+**`JSONObject`**
+Search the source code for the following keywords:
+
+- `import org.json.JSONObject;`
+- `import org.json.JSONArray;`
 
-**https://www.securecoding.cert.org/confluence/display/java/SER04-J.+Do+not+allow+serialization+and+deserialization+to+bypass+the+security+manager**
+**`GSON`**
+Search the source code for the following keywords:
+
+- `import com.google.gson`
+- `import com.google.gson.annotations`
+- `import com.google.gson.reflect`
+- `import com.google.gson.stream`
+- `new Gson();`
+- Annotations such as: `@Expose`, `@JsonAdapter`, `@SerializedName`,`@Since`, `@Until`
+
+**`Jackson`**
+Search the source code for the following keywords:
+
+- `import com.fasterxml.jackson.core`
+- `import org.codehaus.jackson` for the older version.
+
+##### ORM
+When using an ORM library, verify that the the data is stored in an encrypted database or that the json is indivually encrypted before storing it.  See the chapters on data storage and cryptographic management for more details. 
```

Comment:
Changed it: some ORM libraries use JSON as a class representation in the DB. But will leave that out.